pkg/chunk/object/chunk_object_client.go

```diff
@@ -0,0 +1,80 @@
+package object
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+
+	"github.com/cortexproject/cortex/pkg/chunk"
+	"github.com/cortexproject/cortex/pkg/chunk/util"
+)
+
+// ChunkClient is used to store chunks in object store backends
+type ChunkClient struct {
+	Store chunk.ObjectClient
+}
+
+// Stop shuts down the object store and any underlying clients
+func (o *ChunkClient) Stop() {
+	o.Store.Stop()
+}
+
+// PutChunks stores the provided chunks in the configured backend. If multiple errors are
+// returned, the last one sequentially will be propagated up.
+func (o *ChunkClient) PutChunks(ctx context.Context, chunks []chunk.Chunk) error {
+	var (
+		chunkKeys []string
+		chunkBufs [][]byte
+	)
+
+	for i := range chunks {
+		buf, err := chunks[i].Encoded()
+		if err != nil {
+			return err
+		}
+		key := chunks[i].ExternalKey()
+
+		chunkKeys = append(chunkKeys, key)
+		chunkBufs = append(chunkBufs, buf)
+	}
+
+	incomingErrors := make(chan error)
+	for i := range chunkBufs {
+		go func(i int) {
+			incomingErrors <- o.Store.PutObject(ctx, chunkKeys[i], bytes.NewReader(chunkBufs[i]))
+		}(i)
+	}
+
+	var lastErr error
+	for range chunkKeys {
+		err := <-incomingErrors
+		if err != nil {
+			lastErr = err
+		}
+	}
+	return lastErr
+}
+
+// GetChunks retrieves the specified chunks from the configured backend
+func (o *ChunkClient) GetChunks(ctx context.Context, chunks []chunk.Chunk) ([]chunk.Chunk, error) {
+	return util.GetParallelChunks(ctx, chunks, o.getChunk)
+}
+
+func (o *ChunkClient) getChunk(ctx context.Context, decodeContext *chunk.DecodeContext, c chunk.Chunk) (chunk.Chunk, error) {
+	readCloser, err := o.Store.GetObject(ctx, c.ExternalKey())
+	if err != nil {
+		return chunk.Chunk{}, err
```

Comment:
In some clients we were doing `errors.WithStack()` on errors. Is it something we wanna keep?

Reply:
I will add this functionality